airflow-core/src/airflow/models/connection.py

```diff
@@ -598,3 +598,13 @@ def get_team_name(connection_id: str, session=NEW_SESSION) -> str | None:
             .where(Connection.conn_id == connection_id)
         )
         return session.scalar(stmt)
+
+    @staticmethod
+    @provide_session
+    def get_bulk_team_name(connection_ids: list[str], session=NEW_SESSION) -> dict[str, str | None]:
```

Comment:
I think this is a confusing method name. It's really getting a connection to team name mapping. It could be more generally useful in other instances, and the current name is somewhat hard coded to the current usage.